File: apps/demo-ios/Sources/Networking/DeviceIdentity.swift
import Foundation

/// A stable per-install device identifier, generated once and persisted in
/// `UserDefaults.standard`. The real gateway's hop 1 (`GET /authorization`)
/// requires an `X-Device-ID` header; this is a demo-harness-only concern and
/// has no bearing on SEACore's own security posture.
enum DeviceIdentity {
    private static let key = "sea.demo.deviceId"

    /// Reused across launches — generated once per install, not per request.
    static var current: String {
        let defaults = UserDefaults.standard
        if let existing = defaults.string(forKey: key) {
            return existing
        }
        let generated = UUID().uuidString
        defaults.set(generated, forKey: key)
        return generated
    }
}

File: apps/demo-rn/src/deviceIdentity.ts
/**
 * A per-process device identifier for the real gateway's `X-Device-ID`
 * header (mirrors apps/demo-ios/Sources/Networking/DeviceIdentity.swift).
 *
 * Simplification vs. the iOS demo: generated once per app launch rather
 * than persisted (no AsyncStorage/UserDefaults-equivalent dependency in
 * this bridge-validation harness) — fine here since this header is a
 * demo-harness-only concern with no bearing on SEACore's own security
 * posture.
 */
let cached: string | undefined;

export function currentDeviceId(): string {
  if (!cached) {
    cached = `rn-${Date.now().toString(36)}-${Math.random().toString(36).slice(2)}`;
  }
  return cached;
}
